### PR TITLE
Convert RulesUI to work based on RulesComponent

### PR DESCRIPTION
**Pull request by fago**
See https://www.drupal.org/node/2659028

WIP

### CODE REVIEW

File: src/Entity/ReactionRuleConfig.php [hunk 1]
```diff
@@ -160,6 +160,36 @@ public function getExpression() {
...
+
+  /**
+   * Sets the Rules component containing the configuration to be stored.
+   *
+   * @param \Drupal\rules\Engine\RulesComponent $component
+   *   The component.
+   *
+   * @return $this
+   */
+  public function setComponent(RulesComponent $component) {
```
Comment (klausi) [inline src/Entity/ReactionRuleConfig.php]: setComponent()? But no component is set in the implementation? Should this be setComponentConfiguration() or transferComponentConfig() or similar?

Comment (fago) [inline src/Entity/ReactionRuleConfig.php]: true, the name might be misleading - maybe setByComponent ?

**Comment by klausi** [comment]
looks like you mixed up the pull request and pushed some unrelated changes?

**Comment by klausi** [comment]
Aha, maybe this is the correct pull request, fixed the title and added a link to the d.o issue.

File: rules.services.yml [hunk 2]
```diff
@@ -34,8 +34,13 @@ services:
...
     tags:
       - { name: 'event_subscriber' }
+  rules.ui_route_enhancer:
+    class: Drupal\rules\Routing\RulesUiRouteEnhancer
+    arguments: ['@plugin.manager.rules_ui']
+    tags:
+      - { name: route_enhancer }
   rules.ui_route_subscriber:
-    class: Drupal\rules\Routing\UiRouteSubscriber
+    class: Drupal\rules\Routing\RulesUiRouteSubscriber
```
Comment (klausi) [inline rules.services.yml:43]: why the name change? No need for the "Rules" prefix?

File: src/Entity/ReactionRuleConfig.php (hunk 1 above)
Comment (klausi) [inline src/Entity/ReactionRuleConfig.php]: The point is that config is transfered over, so should "config" be in the name?

File: rules.services.yml (hunk 2 above)
Comment (fago) [inline rules.services.yml:43]: it's a bit more consistent with the other routing class (enhancer), the RulesUI term and eases finding the subscriber when doing class lockups. I stumbled over it multiple times.

**Comment by fago** [comment]
That's merged.

**Pull request closed without merging**